python/setup.py

```diff
@@ -101,7 +101,7 @@ def initialize_options(self):
             os.environ.get('PYARROW_WITH_PARQUET', '0'))
         self.with_plasma = strtobool(
             os.environ.get('PYARROW_WITH_PLASMA', '0'))
-        if self.with_plasma:
+        if self.with_plasma and "plasma" not in self.CYTHON_MODULE_NAMES:
```

Comment:
The function initialize_options seems to have been called twice and plasma was added twice to CYTHON_MODULE_NAMES, which broke the "python setup.py develop". If there is a better fix let me know!